ilastik/workflows/pixelClassification/pixelClassificationWorkflow.py

```diff
@@ -264,9 +258,15 @@ def handleAppletStateUpdateRequested(self):
         self._shell.setAppletEnabled(self.dataSelectionApplet, not live_update_active and not batch_processing_busy)
         self._shell.setAppletEnabled(self.featureSelectionApplet, input_ready and not live_update_active and not batch_processing_busy)
         self._shell.setAppletEnabled(self.pcApplet, features_ready and not batch_processing_busy)
-        self._shell.setAppletEnabled(self.dataExportApplet, predictions_ready and not batch_processing_busy)
+        self._shell.setAppletEnabled(self.dataExportApplet, features_ready and not batch_processing_busy)
 
         if self.batchProcessingApplet is not None:
+            predictions_ready = features_ready and \
+                                not invalid_classifier and \
+                                len(opDataExport.Inputs) > 0 and \
+                                opDataExport.Inputs[0][0].ready() and \
+                                (TinyVector(opDataExport.Inputs[0][0].meta.shape) > 0).all()
```

Comment:
will this work with multiple lanes?

Reply:
I've just copied it from above, since now it `predictions_ready` only needs to be evaluated for the batch processing stage. However I'm not sure I understood what you mean: are you suggesting we should rather iterate through all of the `Inputs` (higher level slots) and check the condition instead of just checking the first one `Inputs[0]`?